Java.md

```diff
@@ -1 +1,201 @@
-# Java
\ No newline at end of file
+# Java
+
+## Class
+
+### Layout
+
+Methods and fields should be organized as below:
+
+```java
+class MySuperbClass {
+	// Fields
+	
+	// Constructors
+	
+	// Private & protected methods
+	
+	// Public methods
+}
+```
+
+### Indentation
+
+Standard indentation should be **4 soft spaces**. When a carriage return is needed, keep same indentation.
+
+```java
+class AnotherAmazingClass {
+    public void aMethod() {
+        anotherMethod(
+            firstArg,
+            secondArg,
+            thirdArg,
+            // ...
+        );
+        
+        stillAnotherMethod();
+    }
+}
+```
+
+### Names
+
+Class and method names should be the most limpid as possible. Do not worry about having a long class name. Auto-completion and refactor tools are there for helping you.
+
+Prefer `sortUsersByNameAsc` to `getUsers`.
+
+### Case
+
+In Java world, methods and vars should use **camel case**: `myFantasticMethod` . Class should be **pascal case**: `FriendlyClass`.
+
+Besides, it is a good practice to prefix private and protected methods and fields **by an underscore**. In no time at all, you can easily spot the access of your field/method.
```

Comment:
This is also good for when we're instrumenting classes for testing, where we have to make a private method public, the underscore still indicates that it's internal.